tests/ttnn/unit_tests/operations/test_moreh_getitem.py

```diff
@@ -655,3 +717,81 @@ def test_getitem_tilized_five_indices(shape_index_dims, dtype, index_size, row_m
     logger.info(out)
 
     assert passing
+
+
+@skip_for_blackhole("Mismatching on Blackhole, see #12349")
+@pytest.mark.parametrize(
+    "shape_index_dim",
+    (((7, 70), 0),),
+    ids=[
+        "2d_W",
+    ],
+)
+@pytest.mark.parametrize(
+    "dtype",
+    (
+        torch.int32,
+        torch.bfloat16,
+    ),
+    ids=["int32", "bfloat16"],
+)
+@pytest.mark.parametrize(
+    "index_size",
+    (4,),
+)
+@pytest.mark.parametrize(
+    "row_major_index",
+    (False,),
+)
+def test_getitem_tilized_one_index_callback(
+    shape_index_dim, dtype, index_size, row_major_index, device, use_program_cache
+):
+    shape, index_dim = shape_index_dim
+    torch.manual_seed(2)
+
+    if dtype == torch.int32:
+        tt_dtype = ttnn.int32
+    if dtype == torch.bfloat16:
+        tt_dtype = ttnn.bfloat16
+
+    x = torch.randint(low=0, high=10, size=shape).to(dtype)
+
+    dev_x = ttnn.Tensor(x, tt_dtype).reshape(shape).pad_to_tile(float("nan")).to(ttnn.TILE_LAYOUT).to(device)
+
+    idx_value_max = shape[index_dim] - 1
+    idx = torch.randint(-idx_value_max - 1, idx_value_max, (index_size,))
+    if row_major_index:
+        dev_idx = ttnn.Tensor(idx, ttnn.int32).to(device)
+    else:
+        dev_idx = (
+            ttnn.Tensor(idx, ttnn.int32)
+            .reshape([1, 1, 1, 1, index_size])
+            .pad_to_tile(float("nan"))
+            .to(ttnn.TILE_LAYOUT)
+            .to(device)
+        )
+
+    if index_dim == 0:
+        tt_cpu = x[idx]
+    elif index_dim == 1:
+        tt_cpu = x[:, idx]
+    elif index_dim == 2:
+        tt_cpu = x[:, :, idx]
+    elif index_dim == 3:
+        tt_cpu = x[:, :, :, idx]
+    elif index_dim == 4:
+        tt_cpu = x[:, :, :, :, idx]
+
+    for _ in range(2):
+        tt_npu = ttnn.operations.moreh.getitem(dev_x, [dev_idx], [index_dim])
```

Comment:
The current approach uses the same input and output tensors in the second loop. This could allow the test to pass without updating the tensor address in the callback implementation. 

To avoid this, it would be better to move the tensor creation entirely inside the for loop.